MapboxNavigation/RouteVoiceController.swift

```diff
@@ -162,22 +162,34 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
     }
     
     func duckAudio() throws {
+        let audioSession = AVAudioSession.sharedInstance()
         if #available(iOS 12.0, *) {
-            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, mode: AVAudioSessionModeVoicePrompt, options: [.duckOthers, .mixWithOthers])
+            try audioSession.setCategory(.playback, mode: .voicePrompt, options: [.duckOthers, .mixWithOthers])
+        } else if #available(iOS 10.0, *) {
+            try audioSession.setCategory(.ambient, mode: .spokenAudio, options: [.duckOthers, .mixWithOthers])
         } else {
-            try AVAudioSession.sharedInstance().setMode(AVAudioSessionModeSpokenAudio)
-            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, with: [.duckOthers, .mixWithOthers])
+            try audioSession.setMode(.spokenAudio)
+            audioSession.perform(Selector("setCategory:withOptions:error:" as String),
+                                 with: AVAudioSession.Category.ambient.rawValue,
+                                 with: [AVAudioSession.CategoryOptions.duckOthers.rawValue,
+                                        AVAudioSession.CategoryOptions.mixWithOthers.rawValue])
         }
-        try AVAudioSession.sharedInstance().setActive(true)
+        try audioSession.setActive(true)
     }
     
     func mixAudio() throws {
-        try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryAmbient)
-        try AVAudioSession.sharedInstance().setActive(true)
+        let audioSession = AVAudioSession.sharedInstance()
+        if #available(iOS 10.0, *) {
+            try audioSession.setCategory(.ambient, mode: audioSession.mode)
+        } else {
+            audioSession.perform(Selector("setCategory:error:" as String),
+                                 with: AVAudioSession.Category.ambient.rawValue)
```

Comment:
This is necessary because Swift 4.2 obsoleted `setCategory(_:)` in favor of `setCategory(_:mode:)`, which isn’t supported on iOS 9. The same workaround was used in AudioKit/AudioKit#1554.